odoo/addons/base/models/ir_actions_report.py

```diff
@@ -413,27 +338,45 @@ def _run_wkhtmltopdf(
         os.close(pdf_report_fd)
         temporary_files.append(pdf_report_path)
 
+        content = ""
         try:
-            wkhtmltopdf = [_get_wkhtmltopdf_bin()] + command_args + files_command_args + paths + [pdf_report_path]
-            process = subprocess.Popen(wkhtmltopdf, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-            out, err = process.communicate()
-
-            if process.returncode not in [0, 1]:
-                if process.returncode == -11:
-                    message = _(
-                        'Wkhtmltopdf failed (error code: %s). Memory limit too low or maximum file number of subprocess reached. Message : %s')
-                else:
-                    message = _('Wkhtmltopdf failed (error code: %s). Message: %s')
-                raise UserError(message % (str(process.returncode), err[-1000:]))
+            chrome = ChromeBrowser(_logger, browser_size)
+            chrome._websocket_send('Network.enable')
+            chrome._websocket_send('Runtime.enable')
+            chrome._websocket_send('Page.enable')
+            streams = []
+            params = {
+                'displayHeaderFooter': True,
+                'preferCSSPageSize': True,
+                'printBackground': True,
+            }
+            if len(paths) == 1:
+                params.update({
+                    'headerTemplate': headers[0].decode("utf-8") if headers else "",
+                    'footerTemplate': footers[0].decode("utf-8") if footers else ""
+                })
+                chrome.navigate_to("file://" + paths[0], True)
+                res_id = chrome._websocket_send('Page.printToPDF', params=params)
+                res = chrome._websocket_wait_id(res_id)
+                content = res.get('result', {}).get('data')
+                pdf_content = base64.decodebytes(bytes(content.encode('utf-8')))
             else:
-                if err:
-                    _logger.warning('wkhtmltopdf: %s' % err)
+                for i, path in enumerate(paths):
+                    params.update({
+                        'headerTemplate': headers[i].decode("utf-8"),
+                        'footerTemplate': footers[i].decode("utf-8")
+                    })
+                    chrome.navigate_to("file://" + path, True)
+                    res_id = chrome._websocket_send('Page.printToPDF', params=params)
+                    res = chrome._websocket_wait_id(res_id)
```

Comment:
After analyzing the new code, I see that you still don't run the PDF creations asynchronous. 
We have done some testing with wkhtml2pdf and lunching the PDF creation asynchronous improves a lot the time to generate a PDF with multiple Odoo records.

If the `_websocket_wait_id` was done outside the loop, with the list of processes, the PDF generation would be a lot faster when printing multiple records.